evaluator/builtin/string.go

```diff
@@ -188,3 +191,85 @@ func builtinReplace(args []interface{}, _ context.Context) (interface{}, error)
 
 	return strings.Replace(str, oldStr, newStr, -1), nil
 }
+
+// See: https://dev.mysql.com/doc/refman/5.7/en/cast-functions.html#function_convert
+func builtinConvert(args []interface{}, _ context.Context) (interface{}, error) {
+	value := args[0]
+	Charset := args[1].(string)
+
+	// Casting nil to any type returns nil
+	if value == nil {
+		return nil, nil
+	}
+	str, ok := value.(string)
+	if !ok {
+		return nil, nil
+	}
+	if strings.ToLower(Charset) == "ascii" {
+		return value, nil
+	} else if strings.ToLower(Charset) == "utf8mb4" {
+		return value, nil
+	}
+
+	encoding, _ := charset.Lookup(Charset)
+	if encoding == nil {
+		return nil, errors.Errorf("unknown encoding: %s", Charset)
+	}
+
+	target, _, err := transform.String(encoding.NewDecoder(), str)
+	if err != nil {
+		log.Errorf("Convert %s to %s with error: %v", str, Charset, err)
+		return nil, errors.Trace(err)
+	}
+	return target, nil
+}
+
+func builtinSubstring(args []interface{}, _ context.Context) (interface{}, error) {
+	// The meaning of the elements of args.
+	// arg[0] -> StrExpr
+	// arg[1] -> Pos
+	// arg[2] -> Len (Optional)
+	str, err := types.ToString(args[0])
+	if err != nil {
+		return nil, errors.Errorf("Substring invalid args, need string but get %T", args[0])
+	}
+
+	t := args[1]
+	p, ok := t.(int64)
+	if !ok {
+		return nil, errors.Errorf("Substring invalid pos args, need int but get %T", t)
+	}
+	pos := int(p)
+
+	length := -1
+	if len(args) >= 3 {
```

Comment:
This make me wondering when will `len(args)` be greater than 3, but actually `len(args)` will never be greater than 3.